pkg/authorization/authorizer/scope/converter.go

```diff
@@ -0,0 +1,141 @@
+package scope
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/authorization/rulevalidation"
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// ScopesToRules takes the scopes and return the rules back.  We ALWAYS add the discovery rules and it is possible to get some rules and and
+// an error since errors aren't fatal to evaluation
+func ScopesToRules(scopes []string, namespace string, clusterPolicyGetter rulevalidation.ClusterPolicyGetter) ([]authorizationapi.PolicyRule, error) {
+	rules := append([]authorizationapi.PolicyRule{}, discoveryRules...)
+
+	errors := []error{}
+	for _, scope := range scopes {
+		found := false
+
+		for prefix, evaluator := range scopeEvaluators {
+			if strings.HasPrefix(scope, prefix) {
+				found = true
+				currRules, err := evaluator(scope, namespace, clusterPolicyGetter)
+				if err != nil {
+					errors = append(errors, err)
+					continue
+				}
+
+				rules = append(rules, currRules...)
+			}
+		}
+
+		if !found {
+			errors = append(errors, fmt.Errorf("no scope evaluator found for %q", scope))
+		}
+	}
+
+	return rules, kutilerrors.NewAggregate(errors)
+}
+
+var discoveryRules = []authorizationapi.PolicyRule{
+	{
+		Verbs: sets.NewString("get"),
+		NonResourceURLs: sets.NewString(
+			// API discovery/negotiation
+			"/api", "/api/*",
+			"/apis", "/apis/*",
+			"/oapi", "/oapi/*",
```

Comment:
Can we make constants out of it, there's a TODO in [master.go](https://github.com/openshift/origin/blob/2a4f294f506743d18bec3d3db26cfe23d16a42cb/pkg/cmd/server/origin/master.go#L111-L114) to have that prefix being configurable.